simulation/traffic_simulator/include/traffic_simulator/data_type/follow_trajectory.hpp

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TRAFFIC_SIMULATOR__DATA_TYPE__FOLLOW_TRAJECTORY_HPP_
+#define TRAFFIC_SIMULATOR__DATA_TYPE__FOLLOW_TRAJECTORY_HPP_
+
+#include <geometry_msgs/msg/pose.hpp>
+#include <optional>
+#include <scenario_simulator_exception/exception.hpp>
+#include <type_traits>
+
+namespace traffic_simulator
+{
+namespace follow_trajectory
+{
+struct Vertex
+{
+  double time;
+
+  geometry_msgs::msg::Pose position;
```

Comment:
That suggestion is not accepted because const qualifying a data member would make the class not copy-assignable. If you want this class to be immuable, the side that instantiates this class should const qualify it.